path_login.go

```diff
@@ -396,12 +398,48 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 		for _, userIdentity := range resp.Identity.UserAssignedIdentities {
 			principalIDs[convertPtrToString(userIdentity.PrincipalID)] = struct{}{}
 		}
+	default:
+		// in some cases (particularly WIF), a vm/vmss/resource_id might not be provided, in that case
+		// we'll try to authenticate by matching the claim's app_id to the list of managed identities
+		// (see the comment below on that)
+		if claims.AppID == "" {
+			return errors.New("one of vm_name, vmss_name, resource_id, or an app_id JWT claim must be provided")
```

Comment:
I think the app ID claim key is `appid` instead of `app_id`?

Reply:
it is, ty!